pymc/sampling/jax.py

```diff
@@ -289,40 +291,46 @@ def _update_coords_and_dims(
         dims.update(idata_kwargs.pop("dims"))
 
 
-@partial(jax.jit, static_argnums=(2, 3, 4, 5, 6))
 def _blackjax_inference_loop(
-    seed,
-    init_position,
-    logprob_fn,
-    draws,
-    tune,
-    target_accept,
-    algorithm=None,
+    seed, init_position, logprob_fn, draws, tune, target_accept, **adaptation_kwargs
 ):
     import blackjax
 
-    if algorithm is None:
+    algorithm_name = adaptation_kwargs.pop("algorithm", "nuts")
+    if algorithm_name == "nuts":
         algorithm = blackjax.nuts
+    elif algorithm_name == "hmc":
+        algorithm = blackjax.hmc
+    else:
+        raise ValueError("Only supporting 'nuts' or 'hmc' as algorithm to draw samples.")
 
     adapt = blackjax.window_adaptation(
         algorithm=algorithm,
         logdensity_fn=logprob_fn,
         target_acceptance_rate=target_accept,
+        **adaptation_kwargs,
     )
     (last_state, tuned_params), _ = adapt.run(seed, init_position, num_steps=tune)
     kernel = algorithm(logprob_fn, **tuned_params).step
 
-    def inference_loop(rng_key, initial_state):
-        def one_step(state, rng_key):
-            state, info = kernel(rng_key, state)
-            return state, (state, info)
+    def _one_step(state, xs):
+        _, rng_key = xs
+        state, info = kernel(rng_key, state)
+        return state, (state, info)
 
-        keys = jax.random.split(rng_key, draws)
-        _, (states, infos) = jax.lax.scan(one_step, initial_state, keys)
+    progress_bar = adaptation_kwargs.pop("progress_bar", False)
```

Comment:
This should follow the `progress_bar` from `pm.sample`, but I suspect it know comes from a separate argument?

Reply:
Specifically here we could put `progress_bar` in the right place: https://github.com/pymc-devs/pymc/blob/009da35cb608f254cfc82b86513054d1c56ab992/pymc/sampling/mcmc.py#L367

You'll see that the other two methods propagate `progressbar`. Also watch out for the underscore. PyMC doesn't have one, but blackjax seems to.

Reply:
Make sense, thanks!

Also:
```
pymc: progressbar
nutpie: progress_bar
numpyro: progressbar
blackjax: progress_bar
```